src/target/riscv_debug.c

```diff
@@ -646,7 +653,15 @@ static void riscv_hart_discover_triggers(riscv_hart_s *const hart)
 		riscv_csr_write(hart, RV_TRIG_SELECT | RV_CSR_FORCE_32_BIT, &trigger);
 		/* Try reading the trigger info */
 		uint32_t info = 0;
-		if (!riscv_csr_read(hart, RV_TRIG_INFO | RV_CSR_FORCE_32_BIT, &info)) {
+		bool alternate = false;
+		/* Some chips reply ok but returns 0 in the following call (WCH)*/
+		if (!riscv_csr_read(hart, RV_TRIG_INFO | RV_CSR_FORCE_32_BIT, &info))
+			alternate = true;
+		else {
+			if (!info)
+				alternate = true;
+		}
+		if (alternate) {
```

Comment:
Because of what this is doing, we'd advise this can be condensed thanks to the function call being what's known as a sequence point:

```suggestion
		if (!riscv_csr_read(hart, RV_TRIG_INFO | RV_CSR_FORCE_32_BIT, &info) || !info) {
```

Because `info` is initialised to 0 before the read, and because the || guarantees to execute the left hand side before the right in totality, and the call guarantees sequencing, this eliminates a variable and still does the same thing.